metalearners/rlearner.py

```diff
@@ -323,22 +326,51 @@ def predict(
             tau_hat[variant_indices, treatment_variant - 1] = variant_estimates
         return tau_hat
 
+    @copydoc(MetaLearner.evaluate, sep="\n\t")
     def evaluate(
         self,
         X: Matrix,
         y: Vector,
         w: Vector,
         is_oos: bool,
         oos_method: OosMethod = OVERALL,
-    ) -> dict[str, float | int]:
+        scoring: Scoring | None = None,
+    ) -> dict[str, float]:
+        """In the RLearner case, the ``"treatment_model"`` is always evaluated with the
+        :func:`~metalearners.rlearner.r_loss` and the ``scoring["treatment_model"]``
+        parameter is ignored."""
```

Comment:
What if we want to evaluate the treatment model with a different scoring function than the R-loss ? E.g. if the metalearner type is a hyperparameter itself, we might want to use the same evaluation metric across all metalearners and the R-loss favours the RLearner obviously.

Reply:
You are right that this should be possible, I adapted the code and now it's possible to evaluate the treatment model with whichever metric the user wants (which supports sample weights, as the model is trained with sample weights) besides the R-Loss.
[c20ae75](https://github.com/Quantco/metalearners/pull/8/commits/c20ae75321c972763c177bf5051f276336eb8924)